ideas/276-fast-discovery.md

```diff
@@ -0,0 +1,142 @@
+---
+id: 276-fast-discovery
+title: Fast discovery
+status: draft
+created: 2018-06-22
+category: core
+lead-contributor: dshulyak
+contributors:
+        - dshulyak
+exit-criteria: yes
+success-metrics: yes
+clear-roles: yes
+future-iteration: ?
+roles-needed:
+    - Go Developer
+okrs:
+    - "[PO]: Objective: Messaging is reliable"
+    - "[P2]: KR: Cluster can handle 500 concurrent users"
+---
+
+## Preamble
+
+    Idea: #276-fast-discovery
+    Title: Fast discovery
+    Status: draft
+    Created: 2018-06-22
+
+## Summary
+
+Previously we added support for discovery v5 in status client.
+While it allows us to discover peers in under 5 seconds, and consumes on average 50kb of traffic (sum of inbound and outound).
+This picture completely changes with more concurrent clients. If 100 clients joins simultaneously discovery becomes
```

Comment:
Can we also put some details on why is it so? As far as I remember it's because Discovery V5 has hardcoded timeouts to 500ms and is single-threaded.

Reply:
below are the metrics from running connecting 70 peers at the same time. it is definitely not 500ms and perf issue, as there won't be any traffic received.

|   CONTAINER   | DISCOVERY/INBOUND | DISCOVERY/OUTBOUND | P2P/PEERS |
|---------------|-------------------|--------------------|-----------|
| tests_user_29 |             29262 |              10526 |         2 |
| tests_user_20 |             20726 |              14725 |         2 |
| tests_user_26 |             24481 |              14883 |         2 |
| tests_user_4  |             38182 |              26909 |         2 |
| tests_user_37 |             89425 |              99681 |         0 |
| tests_user_25 |             19351 |              19843 |         2 |
| tests_user_13 |             30511 |              27188 |         2 |
| tests_user_43 |            128937 |             113601 |         0 |
| tests_user_68 |            122626 |              94528 |         1 |
| tests_user_22 |             31257 |              11385 |         2 |
| tests_user_12 |             29971 |              17146 |         2 |
| tests_user_67 |             31317 |              11962 |         2 |
| tests_user_51 |             27906 |               9517 |         2 |
| tests_user_8  |             29165 |              16528 |         2 |
| tests_user_60 |             29983 |              30254 |         2 |
| tests_user_53 |             23313 |              19516 |         2 |
| tests_user_18 |            102413 |             100747 |         0 |
| tests_user_57 |             27058 |              14284 |         2 |
| tests_user_16 |             73551 |              90401 |         0 |
| tests_user_50 |             26732 |              16050 |         2 |
| tests_user_56 |             67515 |              42748 |         2 |
| tests_user_33 |             28595 |              14088 |         2 |
| tests_user_58 |             37467 |              44142 |         2 |
| tests_user_63 |             25251 |              11209 |         2 |
| tests_user_65 |             28258 |              13433 |         2 |
| tests_user_10 |             22228 |              16805 |         2 |
| tests_user_11 |             41695 |              25353 |         2 |
| tests_user_46 |             30442 |              14041 |         2 |
| tests_user_66 |             22306 |              14554 |         2 |
| tests_user_49 |            136700 |              93690 |         1 |
| tests_user_54 |             29500 |              13361 |         2 |
| tests_user_6  |             50538 |              33768 |         2 |
| tests_user_32 |             30224 |              14324 |         2 |
| tests_user_0  |             30568 |              21276 |         2 |
| tests_user_30 |             22455 |              16910 |         2 |
| tests_user_31 |             22251 |              12766 |         2 |
| tests_user_19 |             18244 |              15888 |         2 |
| tests_user_17 |             19003 |              14303 |         2 |
| tests_user_21 |             35463 |              34513 |         2 |
| tests_user_64 |             24551 |              10689 |         2 |
| tests_user_34 |             19483 |              16562 |         2 |
| tests_user_45 |             30853 |              12174 |         2 |
| tests_user_47 |             30041 |              23607 |         2 |
| tests_user_41 |             28291 |               9180 |         2 |
| tests_user_9  |             23526 |              10825 |         2 |
| tests_user_1  |             33401 |              22952 |         2 |
| tests_user_5  |             19773 |              20208 |         2 |
| tests_user_42 |             21228 |              17535 |         2 |
| tests_user_15 |             34419 |              18016 |         2 |
| tests_user_69 |             29038 |              13313 |         2 |
| tests_user_61 |             34774 |              20892 |         2 |
| tests_user_39 |            167096 |             113965 |         1 |
| tests_user_36 |             27045 |              12036 |         2 |
| tests_user_52 |            132417 |              60939 |         0 |
| tests_user_35 |             17379 |              16593 |         2 |
| tests_user_24 |             40448 |              32878 |         2 |
| tests_user_59 |             85957 |              70209 |         2 |
| tests_user_7  |             21252 |               9987 |         2 |
| tests_user_55 |             33754 |              14480 |         2 |
| tests_user_3  |             26841 |               9169 |         2 |
| tests_user_28 |             27284 |              21544 |         2 |
| tests_user_2  |             27231 |              15127 |         2 |
| tests_user_27 |             25878 |              24164 |         2 |
| tests_user_48 |             23233 |              18404 |         2 |
| tests_user_62 |            158109 |             144484 |         1 |
| tests_user_40 |            112112 |              74513 |         0 |
| tests_user_23 |             90272 |              77092 |         0 |
| tests_user_44 |             64214 |              51816 |         2 |
| tests_user_14 |            117014 |              75350 |         0 |
| tests_user_38 |             30056 |              17824 |         2 |

Reply:
This table represents the state after how long? Hm, we probably need to dig into Disc V5 anyway and try to resolve this issue. But it's not related to this swarm.

Reply:
around 10s